docs/reference/changelog.rst

```diff
@@ -2,6 +2,17 @@
 Changelog
 *********
 
+4.2.3 (2024-Sep-18)
+-------------------
+
+Application
+===========
+
+- get_arg_or_config now correctly checks the config service if the passed
```

Comment:
Incidentally, I like this idea of adding the changelog entry on the PR with the change. Maybe we should start doing that on the libraries?